Neuroimages_exercise_SVM_on_GM.py

```diff
@@ -0,0 +1,261 @@
+from keras.models import Model, Sequential
+from keras.losses import binary_crossentropy
+from skimage.transform import rescale, resize
+
+
+if __name__=="__main__":
+    file = os.path.abspath('')#Put the current path
+    AD_files = '\**\*AD*.nii'#find all nifti files with AD in the name
+    AD_path = file + AD_files
+    file = os.path.abspath('')
+    AD_files = '\**\*CTRL*.nii'
+    CTRL_path = file + AD_files
+    CTRL_images = []
+    AD_images = []
+    CTRL_data = []
+    AD_data = []
+    
+    for image in glob.glob(os.path.normpath(AD_path), recursive=True):#for loop in order to create a list of images
+        AD_images.append(sitk.ReadImage(image, imageIO = "NiftiImageIO"))
+    for image in glob.glob(os.path.normpath(CTRL_path), recursive=True):
+        CTRL_images.append(sitk.ReadImage(image, imageIO = "NiftiImageIO"))
+    # CTRL_data = [sitk.GetArrayViewFromImage(x) for x in CTRL_images]
+    # CTRL_data_resized = np.array([resize(image, (128, 128,128),anti_aliasing=True) for image in CTRL_data])#resize images for Autoencoder
+    # AD_data = [sitk.GetArrayViewFromImage(x) for x in AD_images]
+    # AD_data_resized = np.array([resize(image, (128, 128,128),anti_aliasing=True) for image in AD_data])
+#%%MAking labels
+    
+    #one way to permutate data but i can do it better with train_test_split
+    #p = np.random.permutation(data_resized.shape[0])#shuffle AD cases with CTRL
+    #data_resized, labels = data_resized[p,:,:,:], labels[p]
+# #%%
+#     x1 = np.linspace(0, CTRL_data_resized[0].shape[0]-1, CTRL_data_resized[0].shape[0])  
+#     y1 = np.linspace(0, CTRL_data_resized[0].shape[1]-1, CTRL_data_resized[0].shape[1])  
+#     z1 = np.linspace(0, CTRL_data_resized[0].shape[2]-1, CTRL_data_resized[0].shape[2])
+#     X, Y, Z = np.meshgrid(x1, y1, z1)#creating grid matrix
+
+    # #%%just to see if the resize is doing well
+    # import plotly
+    # import plotly.graph_objs as go
+    # from plotly.offline import download_plotlyjs, init_notebook_mode#these two lines are needed for graphic creation. (plotly works on ipython so we need to emulate it)
+    # init_notebook_mode()
+    # data_ein=np.einsum('ijk->jki', CTRL_data_resized[0])#here i swap the two directions "x" and "z" in order to rotate the image
+    # fig = go.Figure(data=go.Volume(
+    # x=X.flatten(),
+    # y=Y.flatten(),
+    # z=Z.flatten(),
+    # value=data_ein.flatten(),
+    # isomin=CTRL_data_resized[0].max()/2,#min value of isosurface ATTENTION: A bigger calcoulation time could bring the rendering to a runtime  error if we use the browser option
+    # isomax=CTRL_data_resized[0].max(),
+    # opacity=0.5, # needs to be small to see through all surfaces
+    # surface_count=15, # needs to be a large number for good volume rendering
+    # caps=dict(x_show=False, y_show=False, z_show=False)
+    # ))
+
+    # fig.show(renderer="browser") 

```

Comment:
This part is only a volumetric rendering plot. The plot will appear as a browser tab because in older version of the packages plotly works only if you implement a notebook-like environment.